client/client.go

```diff
@@ -139,21 +145,21 @@ func initClient(clientUrl, username string, options ...Option) *Client {
 	client := &Client{
 		BaseURL:    bUrl,
 		username:   username,
-		httpClient: http.DefaultClient,
 		MOURL:      DefaultMOURL,
 	}
 
 	for _, option := range options {
 		option(client)
 	}
 
-	transport = client.useInsecureHTTPClient(client.insecure)
-	if client.proxyUrl != "" {
-		transport = client.configProxy(transport)
-	}
-
-	client.httpClient = &http.Client{
-		Transport: transport,
+	if client.httpClient == nil {
+		transport = client.useInsecureHTTPClient(client.insecure)
```

Comment:
it looks like if you pass in your own http client - you'll be ignoring all of the client options, correct?

Reply:
Expectation is that user will configure the client.http with proper transport and proxy. TimeOut option in client.http is still valid.